src/utilities/make_state_vector_file.py

```diff
@@ -181,26 +190,321 @@ def make_state_vector_file(
     # Save
     if save_pth is not None:
         print("Saving file {}".format(save_pth))
-        ds_statevector.to_netcdf(
-            save_pth,
-            encoding={
-                v: {"zlib": True, "complevel": 9} for v in ds_statevector.data_vars
-            },
-        )
+        ds_statevector.to_netcdf(save_pth)
 
     return ds_statevector
 
 
+
+def get_country_code(ROI):
+    """
+    Matches the given country to its country code by searching through online database
+
+    Arguments
+        ROI            [str]  : Name of the country specified in config file
+
+    Returns
+        countryCode    [str]  : Three-letter country code (e.g. USA) of the ROI
+    """
+    geodata_url = "https://gadm.org/download_country.html"  # URL of the database for country shapefiles
+
+    try:
+        # Send a GET request to the geodata URL with a timeout of 5 seconds
+        page = requests.get(geodata_url, timeout=5)
+
+        # Create a BeautifulSoup parser object to parse the HTML content
+        parser = BeautifulSoup(page.content, 'html.parser')
+
+        found = False  # Flag to track if the ROI is found in the database
+
+        # Iterate over each 'option' tag in the parsed HTML content
+        for country in parser.find_all('option'):
+            if country.string == ROI:  # Check if the country name matches the given ROI
+                countryInfo = str(country)
+                # Extract the country code from the HTML string
+                countryCode = countryInfo.split("=\"")[1].split("_")[0]
+                print(countryCode)  # Print the country code
+                found = True
+
+        if not found:
+            # If the ROI is not found in the database, print an error message with a link to the download page
+            print("Error: The ROI entered does not match any of the regions in our database. "
+                  "Please visit https://gadm.org/download_country.html")
+            return None
+
+        return countryCode  # Return the country code
+
+    except Timeout:
+        # Handle timeout exception
+        print("Timeout error: The request to the web server timed out.")
+        return None
+
+
+def get_shapefile(countryCode):
+    """
+    Fetches the shapefile corresponding to the ROI and downloads it in the current directory
+
+    Arguments
+        countryCode    [str]  : Three-letter country code (e.g. USA) of the ROI
+
+    Returns
+        shp_name        [str] : Path to the dowloaded shapefile
+    """
+    shp_url = f"https://geodata.ucdavis.edu/gadm/gadm4.1/shp/gadm41_{countryCode}_shp.zip"  # URL for country shapefile
+
+    try:
+        # Send a GET request to the shapefile URL with a timeout of 5 seconds
+        rshp = requests.get(shp_url, allow_redirects=True, timeout=5)
```

Comment:
You said you were getting timeout errors with the downloads? Some of the files are reasonably large, maybe increase the timeout to 30 seconds?